Mod2/Lessons/Week2/TDD.md

```diff
@@ -0,0 +1,152 @@
+# Test Driven Development
+
+## Learning Goals
+* Describe the benefits of TDD
+* Practice a TDD workflow
+* Define the difference between Unit and Integration tests
+
+## Warm Up
+
+## Why Write Tests
+
+It can be especially difficult to get started on a new project or even a new iteration of a project. The essence of testing is asking questions and coming up with difficult answers.
+
+Testing compels you to make hard decisions early, and upfront. This is scary because you are making decisions in a context you don’t understand.
+Testing (especially in the context of TDD) is a discipline tool that forces you to a) be specific about what you are trying to do and b) stay focused on that objective.
+
+Having a robust test suite is a way for us to be good to our future selves; and provides us with several advantages:
+1. _Refactor with Confidence_: When we decide we want to make a change to how we’ve implemented our code, we can make that change while ensuring that we know the code as a whole still works.
+2. _Add New Features with Confidence_: This also allows us to add new features with confidence. Sometimes it’s difficult to know how code we add may impact functionality that we’ve already provided. A test suite tells us when something new we’ve done has broken something else we did before.
```

Comment:
I would add a note about how you **or your teammates** can refactor or add new features with confidence.